### PR TITLE
solution hw-28.1

### PR DESCRIPTION
**Pull request by RomanHreits**

### CODE REVIEW

**Review by DmGrt — APPROVED** [review]
GJ!

File: src/main/java/com/internet/shop/controller/user/LoginController.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        String login = req.getParameter("login");
+        String pwd = req.getParameter("pwd");
+        try {
+            User user = authenticationService.login(login, pwd);
+            HttpSession session = req.getSession();
+            session.setAttribute("user_Id", user.getId());
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/internet/shop/controller/user/LoginController.java]: let's make it constant as in filter

File: src/main/java/com/internet/shop/controller/user/LoginController.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+        } catch (AuthenticationException e) {
+            req.setAttribute("errorMessage", e.getMessage());
+            req.getRequestDispatcher("WEB-INF/views/user/userLogin.jsp").forward(req, resp);
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/internet/shop/controller/user/LoginController.java]:
```suggestion
            req.getRequestDispatcher("/WEB-INF/views/user/userLogin.jsp")
                .forward(req, resp);
```

File: src/main/java/com/internet/shop/controller/user/UserRegistrationController.java [hunk 3]
```diff
@@ -31,8 +31,7 @@ protected void doPost(HttpServletRequest req, HttpServletResponse resp)
         if (password.equals(req.getParameter("repeatPwd"))) {
             User createdUser = userService.create(new User(login, password));
             cartService.create(new ShoppingCart(createdUser.getId()));
-            req.setAttribute("login", createdUser.getLogin().toUpperCase());
-            req.getRequestDispatcher("/WEB-INF/views/user/userMainPage.jsp").forward(req, resp);
+            resp.sendRedirect(req.getContextPath() + "/login");
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/internet/shop/controller/user/UserRegistrationController.java]: user has only created his account, should he pass his data once again?

File: src/main/java/com/internet/shop/security/AuthenticationServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+public class AuthenticationServiceImpl implements AuthenticationService {
+    @Inject
+    private UserService userService;
+
+    @Override
+    public User login(String login, String password) throws AuthenticationException {
+        Optional<User> userOptional = userService.findByLogin(login);
+        if (userOptional.isPresent() && userOptional.get().getPassword().equals(password)) {
+            return userOptional.get();
+        } else {
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/internet/shop/security/AuthenticationServiceImpl.java]: do we need else after return?

File: src/main/java/com/internet/shop/web/filters/AuthenticationFilter.java [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+        }
+        Long userId = (Long) req.getSession().getAttribute(USER_ID);
+        if (userId == null) {
+            resp.sendRedirect(req.getContextPath() + "/registration");
+            return;
+        }
+        try {
+            userService.get(userId);
+        } catch (NoSuchElementException e) {
+            resp.sendRedirect(req.getContextPath() + "/registration");
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/internet/shop/web/filters/AuthenticationFilter.java]: why to registration and not to login?

File: src/main/java/com/internet/shop/controller/user/LoginController.java (hunk 1 above)
Comment (RomanHreits) [reply src/main/java/com/internet/shop/controller/user/LoginController.java]: done

File: src/main/java/com/internet/shop/controller/user/LoginController.java (hunk 2 above)
Comment (RomanHreits) [reply src/main/java/com/internet/shop/controller/user/LoginController.java]: done

File: src/main/java/com/internet/shop/controller/user/UserRegistrationController.java (hunk 3 above)
Comment (RomanHreits) [reply src/main/java/com/internet/shop/controller/user/UserRegistrationController.java]: done

File: src/main/java/com/internet/shop/security/AuthenticationServiceImpl.java (hunk 4 above)
Comment (RomanHreits) [reply src/main/java/com/internet/shop/security/AuthenticationServiceImpl.java]: done

File: src/main/java/com/internet/shop/web/filters/AuthenticationFilter.java (hunk 5 above)
Comment (RomanHreits) [reply src/main/java/com/internet/shop/web/filters/AuthenticationFilter.java]: done

File: src/main/java/com/internet/shop/controller/user/LoginController.java [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        String login = req.getParameter("login");
+        String pwd = req.getParameter("pwd");
+        try {
+            User user = authenticationService.login(login, pwd);
+            HttpSession session = req.getSession();
+            session.setAttribute(USER_ID, user.getId());
+
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/internet/shop/controller/user/LoginController.java]:
```suggestion
```

**Review by dmytro-hryhoruk — APPROVED** [review]
Good job

File: src/main/java/com/internet/shop/controller/user/LoginController.java (hunk 6 above)
Comment (RomanHreits) [reply src/main/java/com/internet/shop/controller/user/LoginController.java]: done

**Pull request merged**